bmi_cfe.py

```diff
@@ -368,14 +379,14 @@ def finalize_mass_balance(self, verbose=True):
 
         self.vol_soil_end = self.soil_reservoir['storage_m']
         
-        self.global_residual  = self.volstart + self.volin - self.volout - self.volend -self.vol_end_giuh
-        self.schaake_residual = self.volin - self.vol_sch_runoff - self.vol_sch_infilt - self.vol_et_from_rain
+        self.global_residual  = self.volstart + self.volin - self.volout - self.volend -self.vol_end_giuh #(?) - vol_in_nash_end? --- This is already excluded from the system.
```

Comment:
I'm not 100% sure, but I think this is still needed here. Since the global residual needs to account for anything that can be left over. Unless you are sure that it will be zero at any time step, we should still subtract from volstart and volin.